src/Core/Components/DataGrid/FluentDataGridCell.razor.cs

```diff
@@ -61,4 +67,23 @@ protected override void OnInitialized()
 
     public void Dispose() => Owner.Unregister(this);
 
+    /// <summary />
+    internal async Task HandleOnCellClickAsync(string cellId)
+    {
+        if (Owner.Cells.TryGetValue(cellId, out var cell))
+        {
+            if (GridContext.Grid.OnCellClick.HasDelegate)
+            {
+                await GridContext.Grid.OnCellClick.InvokeAsync(cell);
+            }
+
+            if (cell != null && cell.CellType == DataGridCellType.Default)
+            {
+                if (Column is SelectColumn<TGridItem> selColumn)
+                {
+                    await selColumn.AddOrRemoveSelectedItemAsync(Item);
```

Comment:
Won't this be a duplicate with the same call in `HandleOnRowClickAsync` (if `RestrictToCheckbox=false`? And generate double selections at the same time?

Reply:
That's what the `@onclick:stopPropagation` in `FluentDataGridCell` is used for. Would probably be cleaner to just add a condition, right?